docs/tutorials/clojure-guide.md

```diff
@@ -0,0 +1,238 @@
+---
+author: ericdallo
+disqus: emacs-lsp
+---
+
+# Configuring `Emacs` as a `Clojure` IDE
+
+Author: [@ericdallo](https://github.com/ericdallo)
+
+In this guide, I will show you how to configure Emacs to develop Clojure
+using: 
+
+- `lsp-mode`, which has many IDE features like **completion**, **renaming**,
+**reference-searching**, **definition-finding**, **code lenses**, and **code actions**
+- [CIDER](https://github.com/clojure-emacs/cider),
+the Emacs package most famous for enabling **REPL-driven development** with Clojure.
+
+## Why should I try LSP for Clojure?
+
+Coming from the Java world, when I started programming for Clojure in Emacs, it was odd to me that 
+for almost everything you needed to do with Clojure, **you needed a REPL running in your project**, and if you changed anything
+in your project, you would need to reload the modified namespace/code to see the changes in the REPL.
+I also
+missed a lot of features popularized by Java IDEs, like cleaning unused imports, 
+viewing the call hierarchy of a method/function, formatting code automatically, checking syntax in real-time, and using code actions that could, for example, add a missing import.
+Ever since I found [clojure-lsp](https://github.com/clojure-lsp/clojure-lsp), which implements most of those features, I have been using it alongside `CIDER` without issues.
+
+While I really like `CIDER` and believe it is one of the best tools for Clojure development,
+you can complement it with `clojure-lsp` to make programming in Clojure even better.
+
+Once it is running (and you are plugged into the REPL), `CIDER` will provide many of the same features that `clojure-lsp` does, so you need to tell Emacs how to resolve conflicts.
+
+Therefore, if you want to have a more *IDE-ish* experience with Emacs, you should read this guide.
+
+## Basic configuration
+
+Here is a bare-bones `lsp-mode` configuration template to get you started with your own `lsp-mode` config, 
+or to try out in a separate one-off session. Please note that Emacs configuration frameworks such as 
+**Spacemacs** or **Doom Emacs** often ship with `lsp-mode` settings of their own; should you be using such a framework, 
+and find that `lsp-mode` doesn't behave as intended, please make sure to follow this tutorial from a clean starting point.
+in your config or you could run in a separate session.
+
+``` emacs-lisp
+(require 'package)
+(add-to-list 'package-archives '("melpa" . "http://melpa.org/packages/") t)
+(package-initialize)
+
+(setq package-selected-packages '(clojure-mode lsp-mode cider yasnippet lsp-treemacs helm-lsp projectile hydra flycheck company avy which-key helm-xref))
```

Comment:
I've noticed that most of those packages are not mentioned in the guide, so I'm not sure you should suggest to the users to install them.

Reply:
Yes, since this should be the basic, I think we should remove hydra, which-key, helm, yasnippet and projectile (sorry @bbatsov 😆 ), WDYT @yyoncho ? we mention those libs in other guides too

Reply:
I'd removed everything that's not covered in the guide explicitly. :-)